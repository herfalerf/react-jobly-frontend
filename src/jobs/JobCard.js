import React, { useState, useContext, useEffect } from "react";
import "./JobCard.css";
import UserContext from "../user/UserContext";

//Displays basic information about an individual job, rendered on the JobCardList which is rendered on the JobList and the CompanyDetail components.

const JobCard = ({ id, salary, equity, title, companyName }) => {
  console.debug("JobCard");

  //job application functions pulled from UserContext
  const { hasAppliedToJob, applyToJob } = useContext(UserContext);
  const [applied, setApplied] = useState();

  //Gets list of jobs user has applied to on component mount and sets the application status accordingly.
  useEffect(
    function updateAppliedStatus() {
      console.debug("JobCard useEffect updateAppliedStatus", "id=", id);

      setApplied(hasAppliedToJob(id));
    },
    [id, hasAppliedToJob]
  );

  //function for applying to a job.
  async function handleApply(evt) {
    if (hasAppliedToJob(id)) return;
    applyToJob(id);
    setApplied(true);
  }
  return (
    <div className="JobCard card">
      {applied}
      <div className="card-body">
        <h6 className="card-title">{title}</h6>
        <p>{companyName}</p>
        <p>
          <small>Salary: {salary}</small>
        </p>
        <p>
          <small>Equity: {equity}</small>
        </p>

        <button
          className="btn btn-danger font-weight-bold text-uppercase float-right"
          onClick={handleApply}
          disabled={applied}
        >
          {applied ? "Applied" : "Apply"}
        </button>
      </div>
    </div>
  );
};

export default JobCard;
