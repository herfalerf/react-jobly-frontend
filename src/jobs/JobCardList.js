import React from "react";
import JobCard from "./JobCard";

//This component renders the JobCard components.  This component is rendered in JobList and CompanyDetail components.  This is a dumb component.

const JobCardList = ({ jobs }) => {
  return (
    <div className="JobCardList">
      {jobs.map((job) => (
        <JobCard
          key={job.id}
          id={job.id}
          title={job.title}
          salary={job.salary}
          equity={job.equity}
          companyName={job.companyName}
        />
      ))}
    </div>
  );
};

export default JobCardList;
