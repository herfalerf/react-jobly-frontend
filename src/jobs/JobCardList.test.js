import React from "react";
import { render } from "@testing-library/react";
import JobCard from "./JobCard";
import JobCardList from "./JobCardList";
import { UserProvider } from "../testUtils";

it("matches snapshot", function () {
  let jobs = [{ title: "CEO", salary: 1000000, equity: 10, id: "testid" }];
  const { asFragment } = render(
    <UserProvider>
      <JobCardList jobs={jobs}>{/* <JobCard item={item} /> */}</JobCardList>
    </UserProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});
