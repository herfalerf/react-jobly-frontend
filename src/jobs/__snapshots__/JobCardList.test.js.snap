// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="JobCardList"
  >
    <div
      class="JobCard card"
    >
      <div
        class="card-body"
      >
        <h6
          class="card-title"
        >
          CEO
        </h6>
        <p />
        <p>
          <small>
            Salary: 1000000
          </small>
        </p>
        <p>
          <small>
            Equity: 10
          </small>
        </p>
        <button
          class="btn btn-danger font-weight-bold text-uppercase float-right"
        >
          Apply
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
